src/Form/matching-types.d.ts

```diff
@@ -93,20 +93,56 @@ type SupportedMainTypes =
     | 'identities'
     | 'unknown'
 
-type SupportedSubTypes =
-    | SupportedMainTypes
+type SupportedTypes =
+    | 'identities.firstName'
+    | 'identities.middleName'
+    | 'identities.lastName'
+    | 'identities.fullName'
+    | 'identities.phone'
+    | 'identities.addressStreet'
```

Comment:
- Perhaps it would be easier to not check the composite type and instead just rely on the switch statements checking sub and main types?
- Perhaps there's an easier way to declare these string narrowings in general that doesn't need the d.ts files?